### PR TITLE
InterlockedAdd -- clarify atomic operation result value

### PR DESCRIPTION
**Pull request by kouvel**
Continuing from https://github.com/dotnet/dotnet-api-docs/pull/5164

## Summary

The word "new" describing the return value seemed odd to me, since the whole point of `Interlocked` is that such values may not still be prevailing. I couldn't settle on a wording that clarifies that the entire notion of "new value stored" is a poor choice here, since lock-free operations are necessarily and inherently only retrospectively (past-looking) correct. Perhaps *...value that was stored at the time the operation was applied.*?

So I was also trying to figure out a concise way to mention that, if you need to know the stored value *to which this operation was atomically applied*, the only way to recover that is by subtracting the value you supplied from the return value, that is, by reversing your operation, after the fact.

### CODE REVIEW

**Comment by kouvel** [comment]
CC @glenn-slayden

**Comment by kouvel** [comment]
The original PR was already reviewed in https://github.com/dotnet/dotnet-api-docs/pull/5164, the latest commit has review changes

**Comment by kouvel** [comment]
CC @mangod9 for approval

**Review by mangod9 — APPROVED** [review]

**Comment by kouvel** [comment]
@gewarren it seems I can't merge this PR even after the approval above. Any ideas?

**Comment by gewarren** [comment]
@kouvel I think it's because mangod9 doesn't have write perms on this repo. If you were to approve it, you'd be able to merge it.

**Comment by kouvel** [comment]
It's not letting me approve my own PR. Would you be able to approve?

**Review by gewarren — APPROVED** [review]
Oops, sorry, I missed that you were the author.

**Comment by kouvel** [comment]
Thank you!

**Comment by glenn-slayden** [comment]
LGTM, nice work everyone.

**Pull request merged**
